File: LinkedListUtil.py
from typing import List


class ListNode:
    """Definition for singly-linked list."""

    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class LinkedLists:
    """Collection of linked list utilities"""

    @staticmethod
    def create_linked_list(arr: List) -> ListNode:
        """Method to create a linked list from a List"""
        head = ListNode('head')
        next_head = head
        for element in arr:
            next_head.next = ListNode(element)
            next_head = next_head.next
        return head.next

    @staticmethod
    def print_linked_list(head: ListNode) -> None:
        """Method to print the contents of a linked list"""
        output = []
        while head:
            output.append(str(head.val))
            head = head.next
        print(' -> '.join(output))

File: Previous Entries/LinkedListUtil.mjs
class ListNode {
  constructor(val) {
    this.val = val;
    this.next = null;
  }
}

export default class LinkedListUtil {
  createList(array) {
    let head = new ListNode('head');
    let nextHead = head;
    for (let i = 0; i < array.length; i += 1) {
      nextHead.next = new ListNode(array[i])
      nextHead = nextHead.next;
    }
    return head.next;
  }

  printList(linkedList) {
    const output = [];
    while (linkedList) {
      output.push(linkedList.val);
      linkedList = linkedList.next;
    }
    console.log(output.join(' -> '));
  }
}
